### PR TITLE
Enhance default detection

### PR DESCRIPTION
**Pull request by pdanpdan**
- process default on all structure types
- process complex array types (array of arrays/enums/objects)
- add tests for 100% coverage

### CODE REVIEW

**Comment by pdanpdan** [comment]
As this changes parameters for the internal functions I made the PR based on the https://github.com/thedumbterminal/jsonschema-avro/pull/25 PR.
If there are changes required for https://github.com/thedumbterminal/jsonschema-avro/pull/26 I'll rebase this after the merge to avoid problems.

**Comment by thedumbterminal** [comment]
If this change is still required can we isolate this PR to just deal with that feature and any other changes to their specific PRs. Thanks for the work on this.

**Comment by pdanpdan** [comment]
Hello
- I have updated/rebased the PR
- I have added more tests
- the only changes to existing tests are:
  - in omit_properties/expected.json I also removed the top level `doc` property to have one more test
  - in all expected outputs I removed the empty `doc` property (which should not be there)

File: src/index.js [hunk 1]
```diff
@@ -82,142 +76,250 @@ jsonSchemaAvro._isArray = (schema) => schema.type === 'array'
...
+  const mappedType = jsonType !== Object(jsonType)
+    ? typeMapping[jsonType]
+    : jsonType
+  if (mappedType === undefined) {
+    throw new Error(`Invalid JSON schema type "${jsonType}" for "${propName}"`)
+  }
+  return mappedType
+}
+
+jsonSchemaAvro._setTypeAndDefault = (
```
Comment (thedumbterminal) [inline src/index.js:288]: I like this approach!

**Review by thedumbterminal — COMMENTED** [review]
Cheers, will take a look through this when I have a moment, thanks

**Comment by pdanpdan** [comment]
No changes, just a rebase to check that all tests pass after change from float to double

**Review by thedumbterminal — APPROVED** [review]
Nice! Thanks for taking the time to put together a comprehensive change

**Pull request merged**